theano/compile/tests/test_pfunc.py

```diff
@@ -608,10 +608,38 @@ def test_givens_replaces_shared_variable2(self):
         a = shared(1., 'a')
         a.default_update = a + 3
         c = a + 10
-        f = pfunc([], c, givens={a: (a + 10)})
 
-        assert f() == 21
-        assert f() == 34
+        # This is not allowed in Theano anymore
+        self.assertRaises(AssertionError, theano.function,
+                          [], c, givens={a: (a + 10)})
+
+    def test_givens_replaces_shared_variables3(self):
+        a = shared(1, name='a')
+        b = shared(2, name='b')
+        c = shared(3, name='c')
+
+        y = a + b + c
+
+        f_allowed = pfunc([], y, givens= [
+            (a, c),
+            (b, c),
+        ])
+        assert f_allowed() == 9
+
+        self.assertRaises(AssertionError, theano.function,
+                          [], y, givens= [(a, a+1)])
+
+        self.assertRaises(AssertionError, theano.function,
+                          [], y, givens= [(a, c),
+                                          (a, b),])
+
+        self.assertRaises(AssertionError, theano.function,
+                          [], y, givens= [(b, a),
+                                          (a, b),])
+
+        self.assertRaises(AssertionError, theano.function,
+                          [], y, givens= [(b, a),
```

Comment:
I test with the givens: [(a,c), (b, a)] would be useful.

(ps I did just a quick review)